gm4/plugins/manifest.py

```diff
@@ -65,35 +62,43 @@ def update_patch(ctx: Context):
 	"""Retrieves manifest from previous build, and increments patch number
 	 	 if there are any changes between last commit and HEAD"""
 	version = os.getenv("VERSION", "1.19")
-	prefix = int(os.getenv("PATCH_PREFIX", 0))
 	release_dir = Path('release') / version
 	manifest_file = release_dir / "meta.json"
 
 	modules = ctx.cache["gm4_manifest"].json["modules"]
 
 	if manifest_file.exists():
-		old_manifest = json.loads(manifest_file.read_text())
-		last_commit = old_manifest["last_commit"]
-		released_modules: list[dict[str, Any]] = old_manifest["modules"]
+		manifest = json.loads(manifest_file.read_text())
+		last_commit = manifest["last_commit"]
+		released_modules: dict[str, dict[str, Any]] = {m["id"]:m for m in manifest["modules"] if m.get("version", None)}
 	else:
 		last_commit = None
-		released_modules = []
+		released_modules = {}
 
-	# Check if there are any changes between last commit and HEAD
-	for module in modules:
-		id = module["id"]
+	for id in modules:
+		module = modules[id]
 
 		diff = run(["git", "diff", last_commit, "--shortstat", "--", id]) if last_commit else True
-		released = next((m for m in released_modules if m["id"] == id), None)
+		released = released_modules.get(id, None)
 
 		if not diff and released:
 			# No changes were made, keep the same patch version
-			module["patch"] = released["patch"]
+			module["version"] = released["version"]
+		elif not released:
+			# First release
+			module["version"] = module["version"].replace("X", "0")
 		else:
-			# Changes were made or this is the first release, bump the patch
-			patch = released["patch"] if released else prefix
-			module["patch"] = patch + 1
-			print(f"[GM4] Updating {id} to {patch + 1}")
+			# Changes were made, bump the patch
+			version = dict(zip(("major", "minor"), map(int, module["version"].split(".")[0:2])))
+			last_ver = dict(zip(("major", "minor", "patch"), map(int, released["version"].split("."))))
+			
+			if version["minor"] > last_ver["minor"]:
```

Comment:
This also needs to check for major version bumps. If the logic gets too complicated there is probably a semver package out there.

Reply:
Oh shoot you're right. Since a major increment necessarily comes with a minor reset, I was originally going to make it whenever the minor version changes be the trigger. But I should just be able to check if major increment or if minor increment